ntbs-integration-tests/ClinicalDetailsPageTests.cs

```diff
@@ -0,0 +1,173 @@
+
+            var formData = new Dictionary<string, string>
+            {
+                ["NotificationId"] = Utilities.DRAFT_ID.ToString(),
+                ["NotificationSiteMap[PULMONARY]"] = "true",
+                ["NotificationSiteMap[OTHER]"] = "true",
+                ["OtherSite.SiteDescription"] = "tissue",
+                ["ClinicalDetails.NoSampleTaken"] = "true",
+                ["ClinicalDetails.BCGVaccinationState"] = "Yes",
+                ["ClinicalDetails.BCGVaccinationYear"] = "2000",
+                ["FormattedPresentationDate.Day"] = "1",
+                ["FormattedPresentationDate.Month"] = "1",
+                ["FormattedPresentationDate.Year"] = "2012",
+                ["ClinicalDetails.IsPostMortem"] = "false",
+                ["ClinicalDetails.IsShortCourseTreatment"] = "true",
+                ["ClinicalDetails.IsMDRTreatment"] = "false",
+            };
+
+            // Act
+            var result = await SendFormWithData(document, formData);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.Redirect, result.StatusCode);
+            Assert.Equal(result.Headers.Location.OriginalString, BuildRoute(Routes.ContactTracing, Utilities.DRAFT_ID));
+
+            response = await client.GetAsync(GetPageRouteForId(Utilities.DRAFT_ID));
+            document = await GetDocumentAsync(response);
+            Assert.True(((IHtmlInputElement)document.GetElementById("NotificationSiteMap_PULMONARY_")).IsChecked);
+            Assert.True(((IHtmlInputElement)document.GetElementById("NotificationSiteMap_OTHER_")).IsChecked);
+            Assert.Equal("tissue", ((IHtmlInputElement)document.GetElementById("OtherSite_SiteDescription")).Value);
+            Assert.True(((IHtmlInputElement)document.GetElementById("ClinicalDetails_NoSampleTaken")).IsChecked);
+            Assert.True(((IHtmlInputElement)GetRadioWithValue(document, "ClinicalDetails_BCGVaccinationState", "Yes")).IsChecked);
+            Assert.Equal("2000", ((IHtmlInputElement)document.GetElementById("ClinicalDetails_BCGVaccinationYear")).Value);
+            Assert.Equal("1", ((IHtmlInputElement)document.GetElementById("FormattedPresentationDate_Day")).Value);
+            Assert.Equal("1", ((IHtmlInputElement)document.GetElementById("FormattedPresentationDate_Month")).Value);
+            Assert.Equal("2012", ((IHtmlInputElement)document.GetElementById("FormattedPresentationDate_Year")).Value);
+            Assert.True(((IHtmlInputElement)GetRadioWithValue(document, "ClinicalDetails_IsPostMortem", "false")).IsChecked);
+            Assert.True(((IHtmlInputElement)GetRadioWithValue(document, "ClinicalDetails_IsShortCourseTreatment", "true")).IsChecked);
+            Assert.True(((IHtmlInputElement)GetRadioWithValue(document, "ClinicalDetails_IsMDRTreatment", "false")).IsChecked);
+        }
+
+        [Fact]
+        public async Task Post_ClearsConditionalInputValues_IfRadiosSetToOtherValue()

```

Comment:
Not sure whether we actually need this as there is a unit test for this already...

Reply:
Belts+braces, so I don't mind it here, too, but I don't see a need to write a test like this for EVERY possible case :)